tests/intelligentDriverVisualization.py

```diff
@@ -20,38 +20,42 @@
 
 circularRoad = Road.CircularRoad()
 
-vehicle1 = IntelligentDriver.IntelligentDriver(s_0=70., v_0=30., delta=4., T=1., a=1., b=1.5)
-vehicle2 = IntelligentDriver.IntelligentDriver(s_0=70., v_0=30., delta=4., T=1., a=1., b=1.5, length=8.)
-vehicle3 = IntelligentDriver.IntelligentDriver(s_0=70., v_0=40., delta=4., T=1., a=1., b=1.5)
+vehicle1 = IntelligentDriver.IntelligentDriver(s_0=7., v_0=26., delta=4., T=1., a=1., b=1.5, length=3., label="car")
+vehicle2 = IntelligentDriver.IntelligentDriver(s_0=7., v_0=10., delta=4., T=1., a=1., b=1.5, length=8., label="truck")
+vehicle3 = IntelligentDriver.IntelligentDriver(s_0=7., v_0=24., delta=4., T=1., a=1., b=1.5)
 
-vehicle4 = IntelligentDriver.IntelligentDriver(s_0=70., v_0=30., delta=4., T=1., a=1., b=1.5)
-vehicle5 = IntelligentDriver.IntelligentDriver(s_0=70., v_0=30., delta=4., T=1., a=1., b=1.5)
-vehicle6 = IntelligentDriver.IntelligentDriver(s_0=70., v_0=30., delta=4., T=1., a=.9, b=1.5)
-vehicle7 = IntelligentDriver.IntelligentDriver(s_0=70., v_0=30., delta=4., T=1., a=1., b=1.5)
+vehicle4 = IntelligentDriver.IntelligentDriver(s_0=7., v_0=20., delta=4., T=1., a=1., b=1.5, label="test")
+vehicle5 = IntelligentDriver.IntelligentDriver(s_0=7., v_0=20., delta=4., T=1., a=1., b=1.5)
+vehicle6 = IntelligentDriver.IntelligentDriver(s_0=7., v_0=20., delta=4., T=1., a=.9, b=1.5)
+vehicle7 = IntelligentDriver.IntelligentDriver(s_0=7., v_0=26., delta=4., T=1., a=1., b=1.5)
 
-lane1 = Lane.SimpleLane()
-lane2 = Lane.SimpleLane()
+length = 300.
+
+lane1 = Lane.SimpleLane(full_length=length)
+lane2 = Lane.SimpleLane(full_length=length)
 
 circularRoad.add_lane(lane1)
 circularRoad.add_lane(lane2)
 
 lane1.add_vehicle(vehicle1)
 lane1.add_vehicle(vehicle2)
-lane1.add_vehicle(vehicle3)
+#lane1.add_vehicle(vehicle3)
 
 lane2.add_vehicle(vehicle4)
-lane2.add_vehicle(vehicle5)
-lane2.add_vehicle(vehicle6)
-lane2.add_vehicle(vehicle7)
+#lane2.add_vehicle(vehicle5)
+#lane2.add_vehicle(vehicle6)
+#lane2.add_vehicle(vehicle7)
+
+circularRoad.initialize_default()
 
-lane1.initialize_default()
-lane2.initialize_default()
+vehicle1.velocity = 20.
+vehicle1.position = 20.
 
```

Comment:
You should probably check at this point that the cars are still ordered, when explicitly setting the position. Otherwise this will break when changing the number of vehicles.

Reply:
Yes, I should remove the explicit change of the position.